opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py

```diff
@@ -226,26 +299,20 @@ def _drain_queue(self):
         `export` that is not thread safe.
         """
         while self.queue:
-            self.export()
+            self.export_batch()
 
     def force_flush(self, timeout_millis: int = 30000) -> bool:
         if self.done:
             logger.warning("Already shutdown, ignoring call to force_flush().")
             return True
 
-        self._flushing = True
-        self.queue.appendleft(self._FLUSH_TOKEN_SPAN)
-
-        # wake up worker thread
         with self.condition:
+            flush_request = self._get_or_create_flush_request()
+            # signal the worker thread to flush and wait for it to finish
             self.condition.notify_all()
 
         # wait for token to be processed
-        with self.flush_condition:
-            ret = self.flush_condition.wait(timeout_millis / 1e3)
-
-        self._flushing = False
-
+        ret = flush_request.event.wait(timeout_millis / 1e3)
```

Comment:
Should we have recalculation of the timeout similar to https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py#L186

Reply:
i don't think it is required. the only thing that might influence the wait time is acquiring the `flush_request` which shouldn't take that much time.